pytorch_lightning/utilities/xla_device.py

```diff
@@ -72,7 +72,8 @@ def _is_device_tpu() -> bool:
         # `xm.get_xla_supported_devices("TPU")` won't be possible.
         if xm.xrt_world_size() > 1:
             return True
-        return len(xm.get_xla_supported_devices("TPU")) > 0
+        tpus = xm.get_xla_supported_devices("TPU")
+        return False if tpus is None else len(tpus) > 0
```

Comment:
This should work for both `None` and `[]`

```suggestion
        return bool(xm.get_xla_supported_devices("TPU"))
```

Reply:
or

```python
return bool(xm.xrt_world_size()) or bool(xm.get_xla_supported_devices("TPU"))
```